controllers/image_generator.go

```diff
@@ -133,6 +134,12 @@ func ImageCreate(c client.Client) (int, error) {
 	}
 
 	ig := GetImageGenerator()
+
+	if ig == nil {
+		igLogger.Info("skipping image creation as ig is nil")
+		return ImageCreationFailed, ErrImageGeneratorNil
+	}
```

Comment:
Have you understand why gets nil? i don't get how the global variable becomes nil after it was initialize

Reply:
So `ig` should be initialised even in case of unsupported provider. And in case of error in initialising ig, the initialisation code should re-run as part of reconcile.